integreat_compass/cms/views/offers/offer_form_view.py

```diff
@@ -0,0 +1,178 @@
+        organization_form = OrganizationForm(instance=offer_organization)
+
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context_data(**kwargs),
+                "offer_form": offer_form,
+                "offer_version_form": offer_version_form,
+                "contact_form": contact_form,
+                "location_form": location_form,
+                "organization_form": organization_form,
+            },
+        )
+
+    # pylint: disable=too-many-locals
+    @transaction.atomic
+    def post(self, request, *args, **kwargs):
+        r"""
+        Submit :class:`~integreat_compass.cms.forms.offers.contact_form.ContactForm`,
+               :class:`~integreat_compass.cms.forms.offers.location_form.LocationForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_form.OfferForm`,
+               :class:`~integreat_compass.cms.forms.offers.offer_version_form.OfferVersionForm` and
+               :class:`~integreat_compass.cms.forms.offers.organization_form.OrganizationForm`
+
+        :param request: The current request
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :raises ~django.core.exceptions.PermissionDenied: If user does not have the permission to edit the specific page
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        offer_instance = Offer.objects.filter(id=kwargs.get("offer_id")).first()
+        if not request.user.has_perm("cms.edit_offer", offer_instance):
+            raise PermissionDenied(
+                f"{request.user!r} does not have the permission to edit {offer_instance!r}"
+            )
+
+        user = User.objects.get(pk=request.user.id)
+
+        offer_form = OfferForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        offer_version_form = OfferVersionForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        contact_form = ContactForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        location_form = LocationForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+        organization_form = OrganizationForm(
+            data=request.POST, additional_instance_attributes={"user": user}
+        )
+
+        forms = (
+            offer_form,
+            offer_version_form,
+            contact_form,
+            location_form,
+            organization_form,
+        )
+        if not all(form.is_valid() for form in forms):
+            for form in forms:
+                form.add_error_messages(request)
+        else:
+            print(contact_form.is_valid())
```

Comment:
Why do you use print here instead of the logger?

Reply:
Also, the printed object itself is meaningless since you are in the `else` case of `all(form.is_valid())`, so we can be 100% sure that `contact_form.is_valid()` returns `True` at this point in time.

Reply:
Sorry, this was just for my own debugging purposes and should already have been removed.

Used print because currently it didn't  make a difference for that purpose since coloring of the logs seems broken at the moment.